libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/MapboxNavigationViewApi.kt

```diff
@@ -0,0 +1,157 @@
+package com.mapbox.navigation.dropin.internal
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.dropin.NavigationViewApi
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.destination.Destination
+import com.mapbox.navigation.ui.app.internal.destination.DestinationAction
+import com.mapbox.navigation.ui.app.internal.endNavigation
+import com.mapbox.navigation.ui.app.internal.extension.dispatch
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterAction
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class MapboxNavigationViewApi(
+    private val store: Store
+) : NavigationViewApi() {
+
+    override fun startFreeDrive() {
+        store.dispatch(endNavigation())
+    }
+
+    override fun startDestinationPreview(point: Point) {
+        store.dispatch(DestinationAction.SetDestination(Destination(point)))
+        store.dispatch(NavigationStateAction.Update(NavigationState.DestinationPreview))
+    }
+
+    override fun startRoutePreview(): Expected<Throwable, Unit> =
```

Comment:
I don't have a strong opinion on `Expected<Throwable, Unit>` vs `Expected<SomeType, SomeTypeError>`. `Expected<Throwable, Unit>` does the job in a sense it returns appropriate error message when required.

However, I don't remember if as a team we made a decision to always return some dataType as a part of `Expected`. @LukasPaczos can you pitch in here?

Reply:
I would consider a full error type to give developers a chance to recover from a fault state.

If we return just a throwable, there's no way for a developer to recover in the runtime, they would have to treat the value as an unknown error and reset the whole experience back to square one. A more explicit type offers us a chance of providing error codes (specific for each of the state-mutating functions), for example, `ERROR_ROUTE_MISSING` for `startNavigation()` that would let the app check whether it has the route reference somewhere, set it and recover, or only in the worst case reset the whole experience. An explicit type also allows us to provide more information about the error that developers could push to their logs if needed, a string message isn't always enough.

That said, if we start with just `Throwable`, we can possibly extend it later if needed without it being a breaking chance. In any case, could we cut a ticket to explore this separately? I don't want this to block the PR.